src/transport/Dat.ts

```diff
@@ -1,92 +1,254 @@
+//         this.src = sharePath;
+//         this.createNode(function(datnode, me:Dat) {
+//             console.log("Created node callback. Joining network.")
+//             me.joinNetwork(datnode, function(datnode, me:Dat) {
+//                 console.log("Join network complete")
+//             });
+//             me.importFiles(datnode, callback);
+//         });
+//     }
 
-    Share(sharePath: string, callback, ignores?: string[]) {
-        this.src = path.join(__dirname, sharePath)
-        let src = this.src
-        var logs = []
-        var parent = this
-        DatNode(this.src, { temp: true }, function (err, dat) {
-            if (err) throw err
-            parent.dat.push(dat)
-            parent.network = dat.joinNetwork()
-            parent.network.once('connection', function () {
-                logs.push("Connected")
-            })
+//     Download(destinationPath: string, key: any, callback) {
+//         this.src =  = destinationPath
+//         if (!fs.existsSync(dest)) fs.mkdirSync(dest)
+//         this.createNode()
+//         return DatNode(ram, { key: key, sparse: true }, function (err, dat) {
+//             if (err) throw err
+//             parent.dat.push(dat)
+//             parent.network = dat.joinNetwork(function(err){
+//                 if (err) throw err
+//                 if (!dat.network.connected || !dat.network.connecting) {
+//                     logs.push('No users currently online for that key.')
+//                     callback(logs)
+//                 }
+//             })
+//             parent.network.once('connection', function () {
+//                 logs.push("Connected")
+//             })
+
+//             dat.archive.metadata.update(download)
+
+//             function download() {
+//                 var progress = mirror({ fs: dat.archive, name: '/' }, dest, function (err) {
+//                     if (err) throw err
+//                     logs.push('Done')
+//                     callback(logs)
+//                     return envelope
+//                 })
+//                 progress.on('put', function (src) {
+//                     logs.push('Downloading', src.name)
+//                 })
+//             }
+//         })
+
+//     }
+
+    // createOutgoingArchive(id:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, {watch:true})
+    // }
+
+    // createTransientOutgoingArchive(id:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, {watch:true, temp:true})
+    // }
+
+    // createIncomingArchive(id:string, key:string, archivePath:string): Promise<DatNode> {
+    //     return this.createDatNode(id, archivePath, { key: key})
+    // }
+
+    // createTransientIncomingArchive(id:string, key:string): Promise<DatNode> {
+    //     return this.createDatNode(id, ram, { key: key, temp: true })
+    // }
+
+    public getDatNode(id:string) {
+        let me = this
```

Comment:
'me' is another old js convention; variants include using 'self'
In this function, is not really needed because there's no callback / delegation going on. It could be removed; it was added for consistency & at one point I thought getDatNode might be promisified.

Reply:
Interesting, in this instance, since this function is quite small. I would personally opt for just the one-liner. That being said, so long as we are consistent, I do not mind it. It personally just feels out of place for me, I can get used to it though.